src/tests/utils/events.cairo

```diff
@@ -70,17 +72,17 @@ pub impl EventSpyExtImpl of EventSpyExt {
         assert!(self.events.len() == 0, "Events remaining on queue");
     }
 
-    fn assert_no_events_left_from(ref self: EventSpy, contract_address: ContractAddress) {
+    fn assert_no_events_left_from(ref self: EventSpy, from_address: ContractAddress) {
         self.fetch_events();
-        assert!(self.count_events_from(contract_address) == 0, "Events remaining on queue");
+        assert!(self.count_events_from(from_address) == 0, "Events remaining on queue");
     }
 
-    fn count_events_from(self: @EventSpy, from: ContractAddress) -> u32 {
+    fn count_events_from(self: @EventSpy, from_address: ContractAddress) -> u32 {
         let mut result: u32 = 0;
         let mut events = self.events.span();
-        while let Option::Some((from_address, _)) = events
+        while let Option::Some((from, _)) = events
             .pop_front() {
```

Comment:
While it doesn't matter in this context, popping for traversing an array is inefficient. We could something like do:

```
    fn count_events_from(self: @EventSpy, from_address: ContractAddress) -> u32 {
        let mut result = 0;
        let mut events = self.events.span();
        let mut remaining = events.len();

        while remaining > 0 {
            let (from, _) = events.at(remaining - 1);
            if from_address == *from {
                result += 1;
            }
            remaining -= 1;
        };
        result
    }
```

Reply:
We do have iterator syntax in the next Cairo release, which should make this easier.